testnet/stacks-node/src/nakamoto_node/miner.rs

```diff
@@ -383,7 +383,7 @@ impl BlockMinerThread {
                                 "block_height" => new_block.header.chain_length,
                                 "consensus_hash" => %new_block.header.consensus_hash,
                             );
-                            return Err(e);
+                            continue;
```

Comment:
This change "fixed" the issue, but I don't think this is exactly what we want. I think we need to do a check to see if we have a new burn block without a sortition here, as well.

This is correctly triggered because the miner proposed a block on the previous burn block's consensus hash, so it basically needs to re-create the proposed block with a new consensus hash. But my guess is this change will break some other behavior, like not shutting down a miner when there _is_ a new sortition.

Reply:
Hm I don't understand. The internal call seems to already do this. Check both the sortition ID and the cosnensus hash before it returns this error. Therefore, it shouldn't continue and should error. not sure what change needs to be made then.

Reply:
Yeah so, this match branch gets bubbled up from `sign_coordinator`, because the miner is gathering signatures for a nakamoto block with burn hash X, but now the burn hash is Y. So, even though we missed a sortition, it's valid to say "well I'm not gonna sign for this block unless you change the burn hash to Y".

Thus the `continue`, because we don't want the miner thread to exit, we just want it to "start a new block from scratch"

Reply:
Turns out it was a different bug..